generate_assets_metadata.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+# Generates the metadata of the assets (ai and wasm). If an asset of the `ai_assets` array
+# contains a `chunk_size` key, the script splits the asset into chunks where each chunk has
+# a maximum size of `chunk_size`. The format of the `chunk_size` value is equivalent to the
+# `SIZE` argument in `split` or `gsplit` on macOS. See `split`/`gsplit` man page for more details.
+#
+# The script needs to be executed in the root of the repository.
+#
+# Usage:
+# ./generate_assets_metadata <path of assets_manifest.json> [<wasm version>] [<path of the wasm output directory>]
+
+OUT_DIR="out"
+ASSETS_METADATA_PATH=$OUT_DIR/assets_metadata.json
+
+if [ -z ${GITHUB_ACTIONS} ]; then
+    if [[ "$OSTYPE" == "darwin"* ]]; then
+        if command gsplit --version /dev/null; then
+            SPLIT="gsplit"
+        else
+            echo "Requires the GNU version of 'split'. Use 'brew install coreutils' to install it."
+            exit 1
+        fi
+    else
+        SPLIT="split"
+    fi
+else
+    if [ $RUNNER_OS == "macOS" ]; then
+        brew install coreutils
+        if command gsplit --version /dev/null; then
+            SPLIT="gsplit"
+        else
+            echo "Cannot find 'gsplit'."
+            exit 1
+        fi
+    else
+        SPLIT="split"
+    fi
+fi
+
+if ! command gomplate -v /dev/null; then
+    echo "Cannot find 'gomplate'."
+    exit 1
+fi
+
+if ! command jq --version /dev/null; then
+    echo "Cannot find 'jq'."
+    exit 1
+fi
+
+split_asset_into_chunks(){
+    local CHUNKS_DIR=$1
+    local ASSET_CHUNK_SIZE=$2
+    local ASSET_PATH=$3
+    local ASSET_FILENAME=$4
+    local ASSET_VERSION=$5
+
+    mkdir ${CHUNKS_DIR}/${ASSET_VERSION}
+
+    $SPLIT --numeric-suffixes=0 -b $ASSET_CHUNK_SIZE $ASSET_PATH "${CHUNKS_DIR}/${ASSET_VERSION}/${ASSET_FILENAME}_${ASSET_CHUNK_SIZE}_"
+}
+
+generate_dart_assets_manifest() {
+    gomplate -d assets_manifest=$ASSETS_METADATA_PATH -f data/asset_templates/assets.dart.tmpl -o bindings/dart/lib/src/common/reranker/assets.dart
+    flutter format bindings/dart/lib/src/common/reranker/assets.dart
+}
+
+calc_checksum() {
+    echo $(shasum -a 256 $1 | awk '{ print $1 }')
+}
+
+gen_ai_assets_metadata() {
+    local ASSET_MANIFEST=$1
+    local TMP_FILE=$OUT_DIR/tmp_file
```

Comment:
We can also use `mktemp` to create a temporary file.

Reply:
good idea 👍